rust/alloc/macros.rs

```diff
@@ -117,6 +117,42 @@ macro_rules! format {
     }}
 }
 
+/// Tries to create a `String` using interpolation of runtime expressions.
+///
+/// The `try_format!` is an equivalent of [`format!`] macro, that will never
+/// panic on memory allocation failure. See [`format!`] for functionality details.
+///
+/// [`format!`]: crate::format
+///
+/// # Errors
+///
+/// If the capacity overflows, or the allocator reports a failure, then an error
+/// is returned.
+///
+/// # Panics
+///
+/// `try_format!` panics if a formatting trait implementation returns an error.
+/// This indicates an incorrect implementation
+/// since `fmt::Write for String` never returns an error itself.
+/// Never panics on memory allocation failure.
+///
+/// # Examples
+///
+/// ```
+/// try_format!("test").unwrap();
+/// try_format!("hello {}", "world!").unwrap();
+/// try_format!("x = {}, y = {y}", 10, y = 30).unwrap();
+/// ```
+#[macro_export]
+#[stable(feature = "kernel", since = "1.0.0")]
+#[cfg_attr(not(test), rustc_diagnostic_item = "try_format_macro")]
```

Comment:
This is not necessary. It isn't referenced within the compiler in amy diagnostic.

Reply:
Fixed